### ISSUE TITLE
ArrayGet should not produce parentheses when used as a store assignment target, e.g. in UPDATE

### ISSUE THREAD

**Issue by skipkolch** [jOOQ/jOOQ#13808]
I need to update the value in an array by index from another array in the same table.
For example:
```
update player
 set item_level[array_position(item_res_ids, '????') ] = ???
 where player = ????
```
I am writing a query using jooq: 
```
 createDSL().update(PLAYER)
        .set(
            arrayGet(
                PLAYER.ITEM_LEVEL,
                field("array_position({0}, {1})", INTEGER, PLAYER.ITEM_RES_IDS, itemResourceID)
            ),
            itemLevel
        )
        .where(PLAYER.PLAYER_ID.eq(playerID))
        .execute();
```
But jooq generates parentheses. As a result, a syntax error occurs.
`update "public"."player" set ("item_level")[array_position("item_res_ids", ?)] = ? where "public"."player"."player_id" = ?`

Tell me please, how can I fix this?

### Versions:
- jOOQ: 3.15.4
- Java: 17
- Database (include vendor): postgress
- OS: win10

**Comment by lukaseder** [comment]
Thanks a lot for your report. Those parentheses were added because a lot of array expressions will require them, see https://github.com/jOOQ/jOOQ/issues/12480. But of course, they're not required if the array expression is a `TableField`. Should be an easy fix / partial revert of #12480.

Will look into this, soon.

**Comment by skipkolch** [comment]
Tried a workaround:

```
createDSL().update(PLAYER)
  .set(
     field(PLAYER.ITEM_LEVEL.getName() + "[array_position({0}, {1})]", INTEGER, PLAYER.ITEM_RES_IDS, itemResourceID),
     itemLevel
   )
   .where(PLAYER.PLAYER_ID.eq(playerID))
   .execute();
```
It seems to have helped

**Comment by lukaseder** [comment]
Fixing this now. I think this isn't specific to `UPDATE`. It applies to any ISO/IEC 9075-2:2016(E) 9.2 Store assignment usage, including e.g. in procedural logic. For example:

```sql
do $$
declare
  v int[] := array[1, 2];
begin 
  v[2] := 3;
  raise info '%', v;
end;
$$
```

This produces `{1,3}`

**Comment by lukaseder** [comment]
I'll handle that procedural thing separately. It's currently not possible to do with `arrayGet()`: https://github.com/jOOQ/jOOQ/issues/13876

Which makes me think whether we're correctly modelling such assignments in general. Re-using the get method seems like a hack...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13877), and 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13878)
